File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>hash;
        for(int i=0;i<nums.size();i++)
        {
            int x=target-nums[i];
            if(hash.count(x))
            {
                return {hash[x],i};
            }
            hash[nums[i]]=i;
        }
        return {};
    }

};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
        HashMap<Integer,Integer>hash=new HashMap<>();
        for(int i=0;i<n;i++)
        {
            int temp=target-nums[i];
            if(hash.containsKey(temp))
            {
                return new int[]{hash.get(temp),i};
            }
            hash.put(nums[i],i);
        }
        return new int[0];
    }
}